### ISSUE TITLE
installcheck tests fail when run by usernames with hyphens

### ISSUE THREAD

**Issue by mattdeluco** [theory/pgtap#216]
Usernames with hyphens must be quoted, the tests fail because they're expecting unquoted usernames.

See attached `regression.diffs` for a complete list.

```
-ok 191 - users_are(users, desc) extras should have the proper diagnostics
+not ok 191 - users_are(users, desc) extras should have the proper diagnostics
+# Failed test 191: "users_are(users, desc) extras should have the proper diagnostics"
+#         have:     Extra users:
+#         "cl-dev"
+#         want:     Extra users:
+#         cl-dev
```

[regression.diffs.txt](https://github.com/theory/pgtap/files/3764012/regression.diffs.txt)
[regression.out.txt](https://github.com/theory/pgtap/files/3764013/regression.out.txt)
[regression.zip](https://github.com/theory/pgtap/files/3764022/regression.zip)

**Comment by nasbyj** [comment]
Hmm, so this is probably worse than it looks; this example happens to break due to a role name, but I suspect there are problems with other identifiers that need to be quoted (such as table names with spaces in them). On top of that, the database case is extra tricky to test since databases and roles are global objects, so having the tests create or drop them is problematic.

**Comment by theory** [comment]
Does this patch fix it?

``` diff
--- a/test/sql/aretap.sql
+++ b/test/sql/aretap.sql
@@ -694,7 +694,7 @@ SELECT * FROM check_test(
 -- Test users_are().
 
 CREATE FUNCTION ___myusers(ex text) RETURNS NAME[] AS $$
-    SELECT COALESCE(ARRAY( SELECT usename FROM pg_catalog.pg_user WHERE usename <> $1 ), '{}'::name[]);;
+    SELECT COALESCE(ARRAY( SELECT quote_ident(usename) FROM pg_catalog.pg_user WHERE usename <> $1 ), '{}'::name[]);;
 $$ LANGUAGE SQL;
 
 SELECT * FROM check_test(
```

**Comment by mattdeluco** [comment]
No, please see attached `regression.diffs`.

It looks like it's now an issue of mismatched levels of quoting (I think):
```
+not ok 182 - users_are(users, desc) should have the proper diagnostics
+# Failed test 182: "users_are(users, desc) should have the proper diagnostics"
+#         have:      Extra users:
+#         "cl-dev"
+#     Missing users:
+#         """cl-dev"""
+#         want: 
```

[regression.diffs.txt](https://github.com/theory/pgtap/files/4650926/regression.diffs.txt)

**Comment by nasbyj** [comment]
The name type might handle this differently than text would...

**Comment by theory** [comment]
Hrm. I ran

``` sql
create user "foo-bar";
```

Then ran the test (without the above patch) and it passed.  How did you create `"cl-dev"`?

**Comment by mattdeluco** [comment]
This is running `make installcheck` as the MacOS user ("cl-dev") in Terminal, using [Postgres.app](https://postgresapp.com) as my Postgres installation - I'm not certain how the user was created.

**Comment by theory** [comment]
What's the output in psql of `\du`?

**Comment by mattdeluco** [comment]
```
cl-dev=# \du
                                   List of roles
 Role name |                         Attributes                         | Member of 
-----------+------------------------------------------------------------+-----------
 cl-dev    | Superuser, Create role, Create DB                          | {}
 postgres  | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
```

**Comment by theory** [comment]
Same for me:

```
david=# \du
Role name |                         Attributes                         | Member of 
-----------+------------------------------------------------------------+-----------
 david     | Superuser, Create role, Create DB                          | {}
 foo-bar   |                                                            | {}
 postgres  | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
```

I guess the difference is that you _run_ the tests as `cl-dev`, yes?

**Comment by mattdeluco** [comment]
Correct, from the command line as user `cl-dev` I run `make installcheck`.

**Comment by theory** [comment]
That was finally what I needed to know to replicate it. Fixed in 007ee2f. Thanks so much for the report!
